src/Components/Component.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <iostream>
+using namespace std;
+template<class queueType>
+class Component {
```

Comment:
Please add comment for the class. See:
https://google.github.io/styleguide/cppguide.html#Class_Comments